pkg/controller/clusterrole/clusterrole_controller.go

```diff
@@ -86,28 +108,37 @@ func (r *ReconcileClusterRole) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	if instance.Name != "cluster-admin" {
-		return reconcile.Result{}, nil
+	if instance.Name != clusterAdminName {
+		// This should never happen because we filter out other cluster roles in the predicates
+		return reconcile.Result{}, fmt.Errorf("received unknown cluster role: %s", instance.Name)
 	}
 
 	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
-		if !utils.ContainsString(instance.ObjectMeta.Finalizers, finalizer) {
-			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, finalizer)
-			if err = r.client.Update(context.Background(), instance); err != nil {
+		if !containsString(instance.ObjectMeta.Finalizers, finalizer) {
```

Comment:
You don't need `containsString` -- `AddFinalizer` already does that check for you. Similar throughout.

Reply:
Without the condition the controller would update the finalizer in every reconcile loop.